### PR TITLE
:arrow_up: update dependencies

### PR DESCRIPTION
**Pull request by jvdd**
Solves #21 

- [x] update & relax dependencies
- [x] create a release candidate: `0.0.8rc1`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,23 +9,19 @@ repository = "https://github.com/predict-idlab/powershap"
...
-numpy = "~1.20"  # To allow python 3.7.x
-pandas = "~1.3"  # To allow python 3.7.x
+python = ">=3.8,<3.10"
 sklearn = "^0.0"
 catboost = "^1.0.5"
 statsmodels = "^0.13.2"
-shap = "^0.40.0"
+numpy = "^1.23.1"
+pandas = "^1.4.3"
+shap = "^0.41.0"
```
Comment (deepandas11) [inline pyproject.toml:18]: Could these changes be a little more flexible in terms of backwards compatibility?
E.g. 
```
numpy ~=1.21
pandas ~=1.4.0
shap ~=0.40.0
```

This keeps them backwards compatible while installing security fixes as and when needed.

Comment (deepandas11) [reply pyproject.toml:18]: The easiest example of a conflict that a hard cap such as this raises would be that of `numba`: 

https://github.com/numba/numba/issues/8242

The planned release is in Jan'23, so any environments that use numba optimizations(fairly common, I'd assume) might not be able to use powershap.

Comment (jvdd) [reply pyproject.toml:18]: Sure! Didn't keep this in mind :see_no_evil:

Comment (jvdd) [reply pyproject.toml:18]: Is the caret requirement not more suitable than the tilde requirement in your example?
https://python-poetry.org/docs/dependency-specification/#caret-requirements

Comment (deepandas11) [reply pyproject.toml]: > Isn't the caret requirement more suitable than the tilde requirement in your example? https://python-poetry.org/docs/dependency-specification/#caret-requirements

Gotcha! Wasn't aware of poetry best practices and the caret option seems more suitable. Thanks for pointing it out.

Comment (deepandas11) [reply pyproject.toml]: Awesome! Thanks for pushing a release candidate. This works out.

Comment (jvdd) [reply pyproject.toml]: No idea.. Just changed it to scikit-learn without any real impact

**Pull request merged**